micropip/manager.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any, List
+
+from micropip import package_index
+
+
+class Manager:
+    """
+    Manager provides an extensible interface for customizing micropip's behavior.
+
+    Each Manager instance holds its own local state that is
+    independent of other instances, including the global state.
+
+    TODO: Implement all of the following global commands to utilize local state.
+    """
+
+    def __init__(self):
+        self.index_urls = package_index.DEFAULT_INDEX_URLS
+
+        # TODO: initialize the compatibility layer
+        self.repodata_packages: dict[str, dict[str, Any]] = {}
```

Comment:
`repodata` is a legacy name. We used a lock file called `repodata.json,` but we've changed its name to `pyodide-lock.json,` so it would probably be better to rename this variable too.

(no action required in this PR)